src/api/api.ts

```diff
@@ -1,65 +1,68 @@
-import { JsonRPCRequest, JsonRPCResponse, SUPPORTED_METHODS } from '@types';
+import { ipcMain, WebContents } from 'electron';
+
+import { IPC_CHANNELS, JsonRPCRequest, JsonRPCResponse, SUPPORTED_METHODS } from '@types';
 import { safeJSONParse } from '@utils';
 
 import { isValidRequest } from './validators';
 
+const toJsonRpcResponse = (response: Omit<JsonRPCResponse, 'jsonrpc'>) => {
+  return { jsonrpc: '2.0', ...response };
+};
+
+const requestSigning = (
+  request: JsonRPCRequest,
+  webContents: WebContents
+): Promise<JsonRPCResponse> => {
+  // @todo Cleaner way of doing this?
+  // @todo Reject?
+  return new Promise((resolve, _reject) => {
+    webContents.send(IPC_CHANNELS.API, request);
+
+    ipcMain.on(IPC_CHANNELS.API, (_event, arg) => {
+      const response = arg as JsonRPCResponse;
+      if (response.id === request.id) {
+        resolve(response);
+      }
+    });
+  });
+};
+
 // Replies follow: https://www.jsonrpc.org/specification
-export const handleRequest = (
+export const handleRequest = async (
```

Comment:
The method is handling both the validation and the switching. 
These are two different responsibilities.
Separate then combine them with pipe.